codex/sales.nim

```diff
@@ -67,37 +69,44 @@ func new*(_: type Sales,
           proving: Proving,
           repo: RepoStore): Sales =
 
-  Sales(context: SalesContext(
-    market: market,
-    clock: clock,
-    proving: proving,
-    reservations: Reservations.new(repo)
-  ))
+  Sales(
+    context: SalesContext(
+      market: market,
+      clock: clock,
+      proving: proving,
+      reservations: Reservations.new(repo),
+      slotQueue: SlotQueue.new()
+    ),
+    subscriptions: SalesSubscriptions.new()
+  )
 
-proc randomSlotIndex(numSlots: uint64): UInt256 =
-  let rng = Rng.instance
-  let slotIndex = rng.rand(numSlots - 1)
-  return slotIndex.u256
+proc remove(sales: Sales, agent: SalesAgent) {.async.} =
+  await agent.stop()
+  if not sales.stopping:
```

Comment:
This had to be added in because the tests would fail with an error like `the sequence was modified while iterating` and then usually followed by a SIGSEGV